examples/kv_store/Cargo.toml

```diff
@@ -8,5 +8,9 @@ edition = "2021"
 [dependencies]
 omnipaxos_core = { path = "../../omnipaxos_core" }
 omnipaxos_storage = { path = "../../omnipaxos_storage" }
+omnipaxos_macros = { path = "../../omnipaxos_macros", optional = true }
```

Comment:
It is common to re-export the macros from within the main crate.

```toml
# omnipaxos_core/Cargo.toml
[features]
# derive/macros
derive = ["omnipaxos_macros"]

[dependencies]
omnipaxos_macros = { path = "../", optional = true }
```

```rust
// omnipaxos_core/lib.rs
// example from serde: https://github.com/serde-rs/serde/blob/v1.0.89/serde/src/lib.rs#L245-L256
#[cfg(feature = "derive")]
#[allow(unused_imports)]
#[macro_use]
extern crate omnipaxos_macros;
#[cfg(feature = "derive")]
#[doc(hidden)]
pub use omnipaxos_macros::*;
```

```toml
[dependencies]
# if not enabled by default
omnipaxos_core = { ... , features = ["derive"] }
```

Reply:
I found re-exporting in the root a bit confusing as the macro would be `omnipaxos_core::Entry` and the actual trait would be `omnipaxos_core::storage::Entry`. 
So I decided to re-export it in a new mod: `omnipaxos_core::macros::Entry`.